src/main/kotlin/org/jetbrains/research/testspark/helpers/psiHelpers/JavaPsiHelper.kt

```diff
@@ -173,145 +136,313 @@ class JavaPsiHelper : PsiHelperInterface {
 
-    /**
-     * Checks if a PSI class is an abstract class.
-     *
-     * @param psiClass the PSI class of interest
-     * @return true if the PSI class is an abstract class, false otherwise
-     */
-    private fun isAbstractClass(psiClass: PsiClass): Boolean {
-        if (psiClass.isInterface) return false
+    override val qualifiedName: String get() = psiClass.qualifiedName ?: ""
+
+    override val text: String get() = psiClass.text
 
-        val methods = PsiTreeUtil.findChildrenOfType(psiClass, PsiMethod::class.java)
-        for (psiMethod: PsiMethod in methods) {
-            if (!isMethodConcrete(psiMethod)) {
-                return true
+    override val fullText: String
+        get() {
+            var fullText = ""
+            val fileText = psiClass.containingFile.text
+
+            // get package
+            packagePattern.findAll(fileText, 0).map {
+                it.groupValues[0]
+            }.forEach {
+                fullText += "$it\n\n"
+            }
+
+            // get imports
+            importPattern.findAll(fileText, 0).map {
+                it.groupValues[0]
+            }.forEach {
+                fullText += "$it\n"
             }
+
+            // Add class code
+            fullText += psiClass.text
+
+            return fullText
         }
 
-        // check if a class is noted as abstract in the text
-        return psiClass.text.replace(" ", "")
-            .contains("abstractclass${psiClass.name}", ignoreCase = true)
+    override val superClass: PsiClassWrapper? get() = psiClass.superClass?.let { JavaPsiClassWrapper(it) }
+
+    override val virtualFile: VirtualFile get() = psiClass.containingFile.virtualFile
+
+    override val containingFile: PsiFile get() = psiClass.containingFile
+
+    override fun searchSubclasses(project: Project): Collection<PsiClassWrapper> {
+        val scope = GlobalSearchScope.projectScope(project)
+        val query = ClassInheritorsSearch.search(psiClass, scope, false)
+        return query.findAll().map { JavaPsiClassWrapper(it) }
     }
 
+    override fun getInterestingPsiClassesWithQualifiedNames(
+        psiMethod: PsiMethodWrapper,
+    ): MutableSet<PsiClassWrapper> {
+        val interestingMethods = mutableSetOf(psiMethod as JavaPsiMethodWrapper)
+        for (currentPsiMethod in allMethods) {
+            if ((currentPsiMethod as JavaPsiMethodWrapper).isConstructor) interestingMethods.add(currentPsiMethod)
+        }
+        val interestingPsiClasses = mutableSetOf(this)
+        interestingMethods.forEach { methodIt ->
+            methodIt.parameterList.parameters.forEach { paramIt ->
+                PsiTypesUtil.getPsiClass(paramIt.type)?.let { typeIt ->
+                    JavaPsiClassWrapper(typeIt).let {
+                        if (!it.qualifiedName.startsWith("java.")) {
+                            interestingPsiClasses.add(it)
+                        }
```

Comment:
Here, the `it.qualifiedName` may be `""` as well; in this case, it will be added to the list. 

I believe classes like `String` will fall into this category; if a class under test has a `String` parameter in one of its methods, this parameter might be added here.